### ISSUE TITLE
Built-in helpers (e.g. `concat`, `array`, `hash`) should never be flagged by `no-implicit-this`

### ISSUE THREAD

**Issue by lolmaus** [ember-template-lint/ember-template-lint#1129]
This is a perfectly valid template:

```hbs
<Foo/>
{{~concat~}}
<Bar/>
```

The `concat` helper should produce no output, since it has no arguments. But the tildes should remove whitespace around it.

It does not matter which helper I use for this purpose, it just has to be a helper that does nothing without arguments. `concat`, though, seems to be giving a hint of what's going on. 🔧✨

This does work.

But the linter complains with:

> Ambiguous path 'concat' is not allowed. Use '@concat' if it is a named argument or 'this.concat' if it is a property on 'this'. If it is a helper or component that has no arguments you must manually add it to the 'no-implicit-this' rule configuration, e.g. 'no-implicit-this': { allow: ['concat'] }.

Obviously, the leading tilde prevents the rule from recognizing that `concat` is a helper.

**Comment by rwjblue** [comment]
The issue here is that no-implicit this cannot tell that `{{concat}}` is a helper (or component) invocation. Built in helpers (like concat) should not be flagged (because we know for certain that it is a helper), but in the general case the guidance in the error message is exactly correct (you have to add helpers that are invoked without arguments to the the rule config for your project).

**Comment by rwjblue** [comment]
Whoops, sorry about that (hit the wrong button).

Keeping this open to ensure that built-in known helpers do not get flagged by `no-implicit-this`.

**Comment by lolmaus** [comment]
Ah, so this is not related to the tilde whitespace removal?

**Comment by rwjblue** [comment]
No, I don't think so, but the best way to check is by making a failing test case 😸

**Comment by lifeart** [comment]
I can't figure out other built in helpers except (debugger) with use-cases without arguments.

**Comment by lolmaus** [comment]
I believe it should be possible to make `ember-template-lint` aware of custom helpers, by configuring the rule.

**Comment by rwjblue** [comment]
@lolmaus - Indeed it is! That is **exactly** what the error message you included in the original issue description tells you to do 😸

**Comment by lolmaus** [comment]
> I can't figure out other built in helpers except (debugger) with use-cases without arguments.

What I found:

```
array
concat
debugger
hash
hasBlock
hasBlockParams
outlet
yield
```

Some of these seems to already have exceptions, e. g. `outlet` and `yield` do not raise the error.

`hasBlock` does not even have any params, even though it's only use case is within `if`.
